app/controllers/Checkout.scala

```diff
@@ -260,18 +262,19 @@ object Checkout extends Controller with LazyLogging with CatalogProvider {
     import views.support.Pricing._
 
     def getAdjustedRatePlans(promo: AnyPromotion): Option[Map[String, String]] = {
-      case class RatePlanPrice(ratePlanId: ProductRatePlanId, plan: DigipackPlan[BillingPeriod])
+      case class RatePlanPrice(ratePlanId: ProductRatePlanId, chargeList: PaidChargeList)
       promo.asDiscount.map { discountPromo =>
-        catalog.planMap.map { case (key, value) => RatePlanPrice(key, value) }.map { ratePlanPrice =>
+        catalog.allSubs.flatten.map(p => RatePlanPrice(p.id, p.charges)).map { ratePlanPrice =>
```

Comment:
I guessed, actually it's not clear what comes out of the flattening otherwise I'd agree